internal/clients/http/image_builder/image_client.go

```diff
@@ -40,70 +45,63 @@ func newImageBuilderClient(ctx context.Context) (clients.ImageBuilder, error) {
 }
 
 func (c *ImageBuilderClient) Ready(ctx context.Context) error {
 	resp, err := c.client.GetReadiness(ctx, headers.AddImageBuilderIdentityHeader)
 	if err != nil {
-		ctxval.Logger(ctx).Error().Err(err).Msgf("Readiness request failed for image builder: %s", err.Error())
+		c.logger.Error().Err(err).Msgf("Readiness request failed for image builder: %s", err.Error())
 		return err
 	}
 	defer resp.Body.Close()
-	if !parsing.IsHTTPStatus2xx(resp.StatusCode) {
-		ctxval.Logger(ctx).Warn().Msgf("Readiness response from image builder: %d", resp.StatusCode)
-		return ClientErr
+
+	err = http.HandleHTTPResponses(ctx, resp.StatusCode)
+	if err != nil {
+		return fmt.Errorf("ready call: %w", err)
 	}
 	return nil
 }
 
 func (c *ImageBuilderClient) GetAWSAmi(ctx context.Context, composeID string) (string, error) {
-	ctxval.Logger(ctx).Info().Msgf("Getting AMI of image %v", composeID)
+	c.logger.Trace().Msgf("Getting AMI of image %v", composeID)
+
 	imageStatus, err := c.fetchImageStatus(ctx, composeID)
 	if err != nil {
 		return "", err
 	}
-	ctxval.Logger(ctx).Debug().Msgf("Verifying AWS type")
+
+	c.logger.Trace().Msgf("Verifying AWS type")
 	if imageStatus.Type != UploadTypesAws {
-		ctxval.Logger(ctx).Warn().Err(err).Msg("Image is not AWS type")
-		return "", BadImageTypeErr
+		return "", fmt.Errorf("%w: expected image type AWS", UnknownImageTypeErr)
 	}
-	awsStatus, ok := imageStatus.Options.(map[string]interface{})
+	ami, ok := imageStatus.Options.(map[string]interface{})["ami"]
 	if !ok {
-		return "", BadImageTypeErr
+		return "", AmiNotFoundInStatusErr
 	}
-	return awsStatus["ami"].(string), nil
+	return ami.(string), nil
 }
 
 func (c *ImageBuilderClient) fetchImageStatus(ctx context.Context, composeID string) (*UploadStatus, error) {
-	ctxval.Logger(ctx).Info().Msgf("Fetching image status %v", composeID)
+	c.logger.Trace().Msgf("Fetching image status %v", composeID)
+
 	resp, err := c.client.GetComposeStatusWithResponse(ctx, composeID, headers.AddImageBuilderIdentityHeader)
 	if err != nil {
-		ctxval.Logger(ctx).Warn().Err(err).Msg("Failed to fetch image status from image builder")
+		c.logger.Warn().Err(err).Msg("Failed to fetch image status from image builder")
 		return nil, fmt.Errorf("cannot get compose status: %w", err)
 	}
-	statusCode := resp.StatusCode()
-	if parsing.IsHTTPNotFound(statusCode) {
-		return nil, ComposeNotFoundErr
-	}
-	if !parsing.IsHTTPStatus2xx(statusCode) {
-		ctxval.Logger(ctx).Warn().Msgf("Image builder replied with unexpected status while fetching image status: %v", statusCode)
-		return nil, ClientErr
-	}
 
-	err = verifyImage(ctx, resp.JSON200)
+	err = http.HandleHTTPResponses(ctx, resp.StatusCode())
 	if err != nil {
-		ctxval.Logger(ctx).Warn().Err(err).Msg("Image status in not ready")
-		return nil, err
+		if errors.Is(err, clients.NotFoundErr) {
+			return nil, fmt.Errorf("fetch image status call: %w", ComposeNotFoundErr)
+		}
+		return nil, fmt.Errorf("fetch image status call: %w", err)
```

Comment:
Wouldn't this be good enough?
```suggestion
		return nil, fmt.Errorf("fetch image status call: %w", err)
```